girder/utility/plugin_utilities.py

```diff
@@ -148,39 +173,45 @@ def loadPlugin(name, root, appconf, apiRoot=None, curConfig=None):
 
 def getPluginDir(curConfig=None):
     """
-    Returns the /path/to the currently configured plugin directory.
+    Returns a set of the /paths/to the currently configured plugin directories.
     """
     if curConfig is None:
         curConfig = config.getConfig()
 
     # This uses the plugin directory specified in the config first.
     if 'plugins' in curConfig and 'plugin_directory' in curConfig['plugins']:
-        pluginsDir = curConfig['plugins']['plugin_directory']
+        pluginsDir = set(curConfig['plugins']['plugin_directory'].split(','))
 
     # If none is specified, it looks if there is a plugin directory next
     # to the girder python package.  This is the case when running from the
     # git repository.
     elif os.path.isdir(os.path.join(ROOT_DIR, 'plugins')):
-        pluginsDir = os.path.join(ROOT_DIR, 'plugins')
+        pluginsDir = set([os.path.join(ROOT_DIR, 'plugins')])
 
     # As a last resort, use plugins inside the girder python package.
     # This is intended to occur when girder is pip installed.
     else:
-        pluginsDir = os.path.join(PACKAGE_DIR, 'plugins')
-    if not os.path.exists(pluginsDir):
-        try:
-            os.makedirs(pluginsDir)
-        except OSError:
-            if not os.path.exists(pluginsDir):
-                print(TerminalColor.warning(
-                    'Could not create plugin directory.'))
-                pluginsDir = None
-    return pluginsDir
+        pluginsDir = set([os.path.join(PACKAGE_DIR, 'plugins')])
+
+    failedPluginDirs = set()
+
+    for pluginDir in pluginsDir:
+        if not os.path.exists(pluginDir):
+            try:
+                os.makedirs(pluginDir)
+            except OSError:
+                if not os.path.exists(pluginDir):
+                    print(TerminalColor.warning(
+                        'Could not create plugin directory %s.' % pluginDir))
+
+                    failedPluginDirs.add(pluginDir)
+
+    return pluginsDir - failedPluginDirs
```

Comment:
My preference would be to return a list here rather than a set.  We might also want to preserve the order provided in the config.